File: USACO_Jan2024/MajorityOpinion.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long
#define pb push_back

void solve(){ // solving each test case

    // inputing N number of cows
    int n; cin >> n;

    // inputing the hay each cow likes
    vector<int> a(n);
    for(int i=0; i<n; i++)
        cin >> a[i];

    // finding the frequency of each hay
    vector<int> cnt(n+1, 0); // 1-indexed 
    for(int i=0; i<n; i++)
        cnt[a[i]]++;

    // finding the maximum frequency
    int mx = *max_element(cnt.begin(), cnt.end());

    // if the maximum frequency is greater than the number of cows, then it is impossible
    int diff = n - mx;
    if(mx - diff > 1){
        cout << "-1\n";
        return;
    }

    vector<int> ans;
    for(int i=1; i<=n; i++){
        if(cnt[i] == mx)
            ans.pb(i);
    }
    sort(ans.begin(), ans.end());

    // print out ans
    for(int i=0; i < ans.size(); i++)
        cout << ans[i] << " ";
    cout << endl;
}

int main(){
    // interating through each test case
    short t;
    cin >> t;
    while(t--)
        solve();
    return 0;
}

File: USACO_Jan2024/MajorityOpinion.py
def solve():
    n = int(input())

    a = list(map(int, input().split()))

    if n == 2:
        if a[0] == a[1]:
            print(a[0])
        else:
            print("-1")
        return
    if n == 1:
        print(a[0])
        return

    ans = set()

    for i in range(n - 2):
        if a[i] in ans and i != n - 3:
            continue

        if a[i] == a[i+1] or a[i] == a[i+2]:
            ans.add(a[i])
            continue
        elif a[i+1] == a[i+2]:
            ans.add(a[i+1])
            continue

    if not ans:
        print("-1")
        return

    print(" ".join(map(str, ans)))


t = int(input())
for _ in range(t):
    solve()
